src/main/java/org/jabref/logic/importer/fileformat/CiteSeerParser.java

```diff
@@ -46,9 +38,23 @@ private BibEntry parseBibEntry(JSONObject jsonObj) throws ParseException {
         bibEntry.setField(StandardField.YEAR, Objects.toString(jsonObj.get("year"), ""));
         bibEntry.setField(StandardField.PUBLISHER, Objects.toString(jsonObj.get("publisher"), ""));
         bibEntry.setField(StandardField.ABSTRACT, Objects.toString(jsonObj.get("abstract"), ""));
-        bibEntry.setField(StandardField.AUTHOR, Objects.toString(Optional.ofNullable(jsonObj.get("authors"))
-                                                                         .orElse(new JSONArray()), ""));
+        bibEntry.setField(StandardField.AUTHOR, parseAuthors(Optional.ofNullable(jsonObj.get("authors")).orElse(new Object())));
         bibEntry.setField(StandardField.JOURNAL, Objects.toString(jsonObj.get("journal"), ""));
+        bibEntry.setField(StandardField.URL, Objects.toString(jsonObj.get("source"), ""));
         return bibEntry;
     }
+
+    private String parseAuthors(Object authorsObj) {
+        String authorsStr = StringUtils.remove(StringUtil.stripBrackets(Objects.toString(authorsObj, "")), '"');
+        return Arrays.stream(authorsStr.split(","))
+                      .reduce(null, (allAuthors, currAuthor) -> {
```

Comment:
You could try use the Authorlist parser from JabRef AuthorList.parse and then do something like getAsLastFirstNamesWithAnd